modules/collect/src/main/java/com/opengamma/collect/result/Result.java

```diff
@@ -46,6 +46,10 @@
  * Functions will be written to have {@code Result<T>} as the return type.
  * Instead of using exceptions, code will return failure results.
  * <p>
+ * Application code using a result should also operate in a functional style.
+ * Use {@link #map(Function)} and {@link #flatMap(Function)} in preference to
+ * {@link #isSuccess()} and {@link #getValue()}.
+ * <p>
```

Comment:
A code example would be good here showing the common usage. Something like:

    Result<Foo> intermediateResult = calculateIntermediateResult();
    return intermediateResult.flatMap(foo -> calculateFinalResult(foo, ...));

and

    Result<Foo> intermediateResult = calculateIntermediateResult();
    return intermediateResult.map(foo -> calculateFinalValue(foo, ...));